builtin/mainmenu/tab_simple_main.lua

```diff
@@ -17,66 +17,79 @@
-		"pwdfield[8.75,2.3;3.5,0.5;te_pwd;]"
-		
-	if render_details then
-		retval = retval .. "tablecolumns[" ..
-			"color,span=3;" ..
-			"text,align=right;" ..                -- clients
-			"text,align=center,padding=0.25;" ..  -- "/"
-			"text,align=right,padding=0.25;" ..   -- clients_max
-			image_column(fgettext("Creative mode"), "creative") .. ",padding=1;" ..
-			image_column(fgettext("Damage enabled"), "damage") .. ",padding=0.25;" ..
-			image_column(fgettext("PvP enabled"), "pvp") .. ",padding=0.25;" ..
-			"color,span=1;" ..
-			"text,padding=1]"                               -- name
-	else
-		retval = retval .. "tablecolumns[text]"
+	-- Update the cached supported proto info,
+	-- it may have changed after a change by the settings menu.
+	common_update_cached_supp_proto()
+	local fav_selected = menudata.favorites[tabdata.fav_selected]
+
+	local retval =
+		"label[9.5,0;".. fgettext("Name / Password") .. "]" ..
+		"field[0.25,3.35;5.5,0.5;te_address;;" ..
+			core.formspec_escape(core.setting_get("address")) .."]" ..
+		"field[5.75,3.35;2.25,0.5;te_port;;" ..
+			core.formspec_escape(core.setting_get("remote_port")) .."]" ..
+		"button[10,2.6;2,1.5;btn_mp_connect;".. fgettext("Connect") .. "]" ..
+		"field[9.8,1;2.6,0.5;te_name;;" ..
+			core.formspec_escape(core.setting_get("name")) .."]" ..
+		"pwdfield[9.8,2;2.6,0.5;te_pwd;]"
+
+
+	if tabdata.fav_selected and fav_selected then
+		if gamedata.fav then
+			retval = retval .. "button[7.7,2.6;2.3,1.5;btn_delete_favorite;" ..
+				fgettext("Del. Favorite") .. "]"
+		end
 	end
-	retval = retval ..
-		"table[-0.05,0;7.55,2.75;favourites;"
 
-	if #menudata.favorites > 0 then
-		retval = retval .. render_favorite(menudata.favorites[1],render_details)
+	retval = retval .. "tablecolumns[" ..
+		image_column(fgettext("Favorite"), "favorite") .. ";" ..
+		"color,span=3;" ..
+		"text,align=right;" ..                -- clients
+		"text,align=center,padding=0.25;" ..  -- "/"
+		"text,align=right,padding=0.25;" ..   -- clients_max
+		image_column(fgettext("Creative mode"), "creative") .. ",padding=1;" ..
+		image_column(fgettext("Damage enabled"), "damage") .. ",padding=0.25;" ..
+		image_column(fgettext("PvP enabled"), "pvp") .. ",padding=0.25;" ..
+		"color,span=1;" ..
+		"text,padding=1]" ..                  -- name
+		"table[-0.05,0;9.2,2.75;favourites;"
 
-		for i=2,#menudata.favorites,1 do
-			retval = retval .. "," .. render_favorite(menudata.favorites[i],render_details)
+	if #menudata.favorites > 0 then
+		local favs = core.get_favorites("local")
+		if #favs > 0 then
+			for i = 1, #favs do
+			for j = 1, #menudata.favorites do
+				if menudata.favorites[j].address == favs[i].address and
+						menudata.favorites[j].port == favs[i].port then
+					table.insert(menudata.favorites, i, table.remove(menudata.favorites, j))
+				end
+			end
+				if favs[i].address ~= menudata.favorites[i].address then
+					table.insert(menudata.favorites, i, favs[i])
+				end
+			end
```

Comment:
I'm just defending a preference and there can't be a violation since nothing is decreed about it in the [guidelines](http://dev.minetest.net/Lua_code_style_guidelines).